providers/flagd/src/main/java/dev/openfeature/contrib/providers/flagd/FlagdProvider.java

```diff
@@ -42,7 +44,18 @@ public FlagdProvider() {
      * @param options {@link FlagdOptions} with
      */
     public FlagdProvider(final FlagdOptions options) {
-        this.flagResolver = new GrpcResolution(options, CacheFactory.getCache(options), this::getState, this::setState);
+        switch (options.getResolverType()) {
+            case IN_PROCESS:
+                this.flagResolver = new InProcessResolver(options, this::setState);
+                break;
+            case FLAGD:
+                this.flagResolver =
+                        new GrpcResolver(options, CacheFactory.getCache(options), this::getState, this::setState);
+                break;
+            default:
+                throw new IllegalStateException(
+                        String.format("Requested unsupported resolver type of %s", options.getResolverType()));
```

Comment:
didn't we agreed only on logging the error here ?

Reply:
This aligns with the OF spec [^1]

`... Exceptions include functions or methods for the purposes for configuration or setup.`

[^1]: https://openfeature.dev/specification/sections/flag-evaluation#requirement-149